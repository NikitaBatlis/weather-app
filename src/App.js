import React from 'react';
import './App.css';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Titles from './components/Titles';
import Forms from './components/Forms';
import Weather from './components/Weather';

const API_KEY = process.env.REACT_APP_WEATHER_API_KEY;

class App extends React.Component {
    //Constructed starting state:
    state = {
        temperature: undefined,
        city: undefined,
        country: undefined,
        humidity: undefined,
        description: undefined,
        icon: undefined,
        error: undefined
    };

    //The getWeather function that will fire and make the call to the API with the city/country inputs from the form component.
    
    getWeather = async (e) => { // 'e' is the props past through from the form component when getWeather function is called.

        e.preventDefault(); //Needed to override default reload on button submit function.

        const city = e.target.city.value; //getting the input value of the city.
        const country = e.target.country.value; //getting input value of the country

        const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${API_KEY}&units=metric`); 
        const data = await api_call.json(); //Parsing the json api_call data.

        if (city && country ) { //if city and country both return true/have inputs.

            if (data.cod === 404) { //if api_call doesn't match any location on the database (404 error), show this error message.
                this.setState({
                    temperature: undefined,
                    city: undefined,
                    country: undefined,
                    humidity: undefined,
                    description: undefined,
                    icon: undefined,
                    error: "Input doesn't match any known location!"
                });
                
            } else { //if inputs match location in database.
                this.setState({
                    temperature: data.main.temp,
                    city: data.name,
                    country: data.sys.country,
                    humidity: data.main.humidity,
                    description: data.weather[0].main,
                    icon: data.weather[0].icon,
                    error: ""
                });
            }

        } else { //if inputs were empty.
            this.setState({
                temperature: undefined,
                city: undefined,
                country: undefined,
                humidity: undefined,
                description: undefined,
                icon: undefined,
                error: "Please enter valid location."
            });
        }
    }

    render() {
        return (
            <div>
                <div className="wrapper">
                    <div className="main">
                        <Container className="container">
                            <Row className="row">

                                <Col className="title-container">
                                    <Titles />
                                </Col>
                                        
                                <Col className="form-container">
                                    <Forms getWeather={this.getWeather} /> 
                                    <Weather 
                                        temperature={this.state.temperature} 
                                        humidity={this.state.humidity}
                                        city={this.state.city}
                                        country={this.state.country}
                                        description={this.state.description}
                                        icon ={this.state.icon}
                                        error={this.state.error}
                                    />
                                </Col>
                                
                            </Row>
                        </Container>
                    </div>
                </div>            
            </div>
        );
    }
}

export default App;