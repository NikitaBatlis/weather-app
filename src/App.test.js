import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test("First snapshot test", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

const city = 'London';
const country = 'UK';
const API_KEY = '9dc25603f879c3b7eaf448686b1c73a5';

test('the data is Response', () => {
  return fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${API_KEY}&units=metric`).then(res => {
    expect(res.status).toBe(200);
  });
});

