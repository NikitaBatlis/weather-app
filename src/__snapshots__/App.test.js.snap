// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<div>
  <div
    className="wrapper"
  >
    <div
      className="main"
    >
      <div
        className="container container"
      >
        <div
          className="row row"
        >
          <div
            className="title-container col"
          >
            <div>
              <h1
                className="title-container__title"
              >
                Weather Finder
              </h1>
              <h3
                className="title-container__subtitle"
              >
                Get the weather conditions for any location!
              </h3>
            </div>
          </div>
          <div
            className="form-container col"
          >
            <div>
              <form
                onSubmit={[Function]}
              >
                <input
                  name="city"
                  placeholder="City..."
                  type="text"
                />
                <input
                  name="country"
                  placeholder="Country..."
                  type="text"
                />
                <button>
                  Get Weather
                </button>
              </form>
            </div>
            <div
              className="weather__info"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
